RecastDemo/Source/main.cpp

```diff
@@ -58,87 +63,92 @@ static SampleItem g_samples[] =
 	{ createTempObstacle, "Temp Obstacles" },
 //	{ createDebug, "Debug" },
 };
-static const int g_nsamples = sizeof(g_samples)/sizeof(SampleItem); 
-
+static const int g_nsamples = sizeof(g_samples) / sizeof(SampleItem);
 
 int main(int /*argc*/, char** /*argv*/)
 {
 	// Init SDL
 	if (SDL_Init(SDL_INIT_EVERYTHING) != 0)
 	{
-		printf("Could not initialise SDL\n");
+		printf("Could not initialise SDL.  Error: %s", SDL_GetError());
 		return -1;
 	}
-	
-	// Center window
-	char env[] = "SDL_VIDEO_CENTERED=1";
-	putenv(env);
 
-	// Init OpenGL
+	// Enable depth buffer.
 	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
 	SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
+	
+	// Set color channel depth.
 	SDL_GL_SetAttribute(SDL_GL_RED_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_GREEN_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_BLUE_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_ALPHA_SIZE, 8);
-//#ifndef WIN32
+	
+	// 4x MSAA.
 	SDL_GL_SetAttribute(SDL_GL_MULTISAMPLEBUFFERS, 1);
 	SDL_GL_SetAttribute(SDL_GL_MULTISAMPLESAMPLES, 4);
-//#endif
 
-	const SDL_VideoInfo* vi = SDL_GetVideoInfo();
+	SDL_DisplayMode displayMode;
+	SDL_GetCurrentDisplayMode(0, &displayMode);
 
 	bool presentationMode = false;
-
-	int width, height;
-	SDL_Surface* screen = 0;
-	
+	Uint32 flags = SDL_WINDOW_OPENGL;
+	int width;
+	int height;
 	if (presentationMode)
 	{
-		width = vi->current_w;
-		height = vi->current_h;
-		screen = SDL_SetVideoMode(width, height, 0, SDL_OPENGL|SDL_FULLSCREEN);
+		// Create a fullscreen window at the native resolution.
+		width = displayMode.w;
+		height = displayMode.h;
+		flags |= SDL_WINDOW_FULLSCREEN;
 	}
 	else
-	{	
-		width = rcMin(vi->current_w, (int)(vi->current_h * 16.0 / 9.0));
-		width = width - 80;
-		height = vi->current_h - 80;
-		screen = SDL_SetVideoMode(width, height, 0, SDL_OPENGL);
+	{
+		float aspect = 16.0f / 9.0f;
+		width = rcMin(displayMode.w, (int)(displayMode.h * aspect)) - 80;
+		height = displayMode.h - 80;
 	}
 	
-	if (!screen)
+	SDL_Window* window;
+	SDL_Renderer* renderer;
+	int errorCode = SDL_CreateWindowAndRenderer(width, height, flags, &window, &renderer);
+
+	if (errorCode != 0 || !window || !renderer)
 	{
-		printf("Could not initialise SDL opengl\n");
+		printf("Could not initialise SDL opengl\nError: %s\n", SDL_GetError());
 		return -1;
 	}
 
-	glEnable(GL_MULTISAMPLE);
+	SDL_SetWindowPosition(window, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED);
+	SDL_GL_CreateContext(window);
 
-	SDL_WM_SetCaption("Recast Demo", 0);
-	
 	if (!imguiRenderGLInit("DroidSans.ttf"))
 	{
-		printf("Could not init GUI renderer.\n");
+		printf("Could not init GUI renderer.");
```

Comment:
Actually this was an accident.  I had added some extra info here attempting to print SDL_GetError() and then realized that imgui has nothing to do with SDL (duh).  When I removed the additional error logging, I accidentally removed the newline as well. :dizzy_face: